### PR TITLE
Refactor s3 script to accept folders

### PR DESCRIPTION
**Pull request by agustito37**
- Now deployment to s3 accepts folders
- Use of async/await notation
- Added static website hosting URL for the bucket
- Removed compression by default, now you have an environment variable `GZIP_ENABLED=false` by default

Fix https://github.com/rootstrap/react-redux-base/issues/151
Fix https://github.com/rootstrap/react-redux-base/issues/170

### CODE REVIEW

File: tools/s3.js [hunk 1]
```diff
@@ -14,76 +17,64 @@ class s3 {
...
-          return;
-        }
+  async clearBucket(bucketObjects) {
+    try {  
+      for (let x = 0; x < bucketObjects.length; x += 1) {
+        const { Key: key } = bucketObjects[x];
+        const deleteParams = { Bucket: this.bucket, Key: key };
+        console.log(chalkProcessing(`Deleting: ${key}`));
+        await this.client.deleteObject(deleteParams).promise();
+      }
```
Comment (bernabe9) [inline tools/s3.js]: Is each iteration waiting for the promise?

File: tools/s3.js [hunk 2]
```diff
@@ -14,76 +17,64 @@ class s3 {
...
-      const folderPath = path.join(__dirname, distPath);
-
-      fs.readdir(folderPath, (err, files) => {
-        if(!files || files.length === 0) {
-          reject(`provided folder '${distPath}' is empty or does not exist.`);
-          return;
-        }
+  async clearBucket(bucketObjects) {
+    try {  
+      for (let x = 0; x < bucketObjects.length; x += 1) {
```
Comment (bernabe9) [inline tools/s3.js]:
```suggestion
      const promises = bucketObjects.map(({ Key })=>
        this.client.deleteObject({ Bucket: this.bucket, Key };).promise();
      );
      await Promises.all(promises)
```

File: tools/s3.js [hunk 3]
```diff
@@ -14,76 +17,64 @@ class s3 {
...
+      throw new Error(`error uploading bucket objects ${err}`);
+    }
+  }
 
-          fs.readFile(filePath, (error, fileContent) => {
-            if (error) { reject(error); }
+  async walkSync (currentDirPath) {
+    const objects = fs.readdirSync(currentDirPath);
+    for(let x = 0; x < objects.length; x += 1) {
+      await this.walkFileOrDir(currentDirPath, objects[x]);
```
Comment (bernabe9) [inline tools/s3.js]: Is this performing sequentially?

**Comment by bernabe9** [comment]
Nice job! It would be great to log the bucket's URL for quick access to the deployed web app.

File: tools/s3.js (hunk 1 above)
Comment (agustito37) [reply tools/s3.js]: yes, but I will remove this if there's no lower quotas.

File: tools/s3.js (hunk 2 above)
Comment (agustito37) [reply tools/s3.js]: Cool, thanks!

File: tools/s3.js (hunk 1 above)
Comment (agustito37) [reply tools/s3.js]: Yes, I'll change this.

File: tools/s3.js (hunk 3 above)
Comment (agustito37) [reply tools/s3.js]: Indeed it is performing sequentially, but I will change this, I want to check AWS quotas before doing the change.

**Comment by agustito37** [comment]
> Nice job! It would be great to log the bucket's URL for quick access to the deployed web app.

I tried that, seems the js is gzipped (it doesn't have the extension) so the deployed app doesn't work. I'll check this.

**Pull request merged**